include/frg/string.hpp

```diff
@@ -308,8 +315,8 @@ class basic_string {
 	}
 
 	int compare(const char *other) const {
-		if(_length != strlen(other))
-			return _length < strlen(other) ? -1 : 1;
+		if(_length != generic_strlen(other))
+			return _length < generic_strlen(other) ? -1 : 1;
```

Comment:
We don't have to compute `generic_strlen` twice here. (Yes, that was already done before, but it's a 3 line fix).

Reply:
oh, I didn't catch that since I mechanically replaced `strlen` with `generic_&`. will fix